fairroot/eventdisplay/xml/FairXMLDetectorColor.cxx

```diff
@@ -0,0 +1,93 @@
+/*
+ * FairXMLDetColor.cxx
+ *
+ *  Created on: 17 lip 2024
+ *      Author: daniel
+ */
+
+#include "FairXMLDetectorColor.h"
+
+#include "FairXMLNode.h"
+
+#include <RtypesCore.h>
+#include <TAttFill.h>
+#include <TGeoNode.h>
+#include <TGeoVolume.h>
+#include <TNamed.h>
+#include <TString.h>
+Int_t FairXMLDetectorColor::ApplyToNode(TGeoNode* node, FairXMLNode* xml) const
+{
+    auto transparency_atrib = xml->GetAttrib("transparency");
+    Char_t transparency = 0;
+    if (transparency_atrib) {
+        if (transparency_atrib->GetValue().Length()) {
+            transparency = transparency_atrib->GetValue().Atoi();
+            node->GetVolume()->SetTransparency(transparency);
```

Comment:
You expect `node` and `xml` to be non-nullptr.

Why not use a reference instead?

```suggestion
Int_t FairXMLDetectorColor::ApplyToNode(TGeoNode& node, FairXMLNode& xml) const
{
    auto transparency_atrib = xml.GetAttrib("transparency");
    Char_t transparency = 0;
    if (transparency_atrib) {
        if (transparency_atrib->GetValue().Length()) {
            transparency = transparency_atrib->GetValue().Atoi();
            node.GetVolume()->SetTransparency(transparency);
```

(I've only rewritten the start as an example.)